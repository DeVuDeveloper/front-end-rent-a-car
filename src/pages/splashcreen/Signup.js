/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prefer-stateless-function */
import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import { BiFastForwardCircle } from 'react-icons/bi';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Modal, Form } from 'antd';
import { signupUser } from '../../actions/auth';
import 'antd/dist/antd.css';
import './login.css';

const Signup = () => {
  const dispatch = useDispatch();
  const { register, handleSubmit } = useForm();
  const history = useHistory();
  const onSubmit = (data) => {
    history.push('/home');
    setTimeout(() => {
      window.location.reload(true);
    }, 1400);
    dispatch(signupUser(data));
  };
  const CollectionCreateForm = ({ visible, onCreate, onCancel }) => {
    const [form] = Form.useForm();
    return (
      <Modal
        visible={visible}
        title="Sign Up"
        cancelText="Cancel"
        onCancel={onCancel}
        onOk={() => {
          form.validateFields().then((values) => {
            form.resetFields();
            onCreate(values);
          });
        }}
      >
        <form
          className="mx-1 mx-md-4"
          onSubmit={handleSubmit(onSubmit)}
          method="post"
        >
          <div className="d-flex flex-row align-items-center mb-4">
            <i className="fas fa-envelope fa-lg me-3 fa-fw" />
            <div className="form-outline flex-fill mb-0">
              <input
                type="name"
                id="form3Example3c"
                className="form-control"
                placeholder="Name"
                {...register('name', { required: true })}
              />
            </div>
          </div>

          <div className="d-flex flex-row align-items-center mb-4">
            <i className="fas fa-envelope fa-lg me-3 fa-fw" />
            <div className="form-outline flex-fill mb-0">
              <input
                type="email"
                id="form3Example3c"
                className="form-control"
                placeholder="Email"
                {...register('email', { required: true })}
              />
            </div>
          </div>

          <div className="d-flex flex-row align-items-center mb-4">
            <i className="fas fa-lock fa-lg me-3 fa-fw" />
            <div className="form-outline flex-fill mb-0">
              <input
                type="password"
                id="form3Example4c"
                className="form-control"
                placeholder="Password"
                {...register('password', { required: true })}
              />
            </div>
          </div>

          <div className="d-flex flex-row align-items-center mb-4">
            <i className="fas fa-key fa-lg me-3 fa-fw" />
            <div className="form-outline flex-fill mb-0">
              <input
                type="password"
                id="form3Example4cd"
                className="form-control"
                placeholder="Password Confirmation"
                {...register('password_confirmation', {
                  required: true,
                })}
              />
            </div>
          </div>

          <div className="d-flex justify-content-center mx-4 mb-3 mb-lg-4">
            <button type="submit" className="btn btn-primary btn-lg">
              Register
            </button>
          </div>
        </form>
      </Modal>
    );
  };

  const CollectionsPage = () => {
    const [visible, setVisible] = useState(false);

    return (
      <div>
        <button
          type="button"
          className="login-btn"
          onClick={() => {
            setVisible(true);
          }}
        >
          <h3>Signup</h3>
          <BiFastForwardCircle />
        </button>
        <CollectionCreateForm
          visible={visible}
          onCancel={() => {
            setVisible(false);
          }}
        />
      </div>
    );
  };
  return (
    <div className="MainDiv">
      <div className="container">
        <CollectionsPage />
      </div>
    </div>
  );
};

export default Signup;
