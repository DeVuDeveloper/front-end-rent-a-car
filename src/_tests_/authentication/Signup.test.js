import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Signup from '../../pages/splashcreen/Signup';
import store from '../../redux/configureStore';

it('Signup renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Signup />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
