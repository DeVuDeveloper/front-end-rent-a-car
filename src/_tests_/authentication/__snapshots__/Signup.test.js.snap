// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<div
  className="MainDiv"
>
  <div
    className="container"
  >
    <div>
      <button
        className="login-btn"
        onClick={[Function]}
        type="button"
      >
        <h3>
          Signup
        </h3>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="m13 16 5-4-5-4zm-6 0 5-4-5-4z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
